src/OpenSSL/SSL.py

```diff
@@ -699,12 +699,22 @@ def load_client_ca(self, cafile):
 
     def set_session_id(self, buf):
         """
-        Set the session identifier.  This is needed if you want to do session
-        resumption.
+        Set the session id to *buf* within which a session can be reused for
+        this Context object.  This is needed when doing session resumption,
+        because there is no way for a stored session to know which Context
+        object it is associated with.
+
+        :param bytes buf: The session id.
 
-        :param buf: A Python object that can be safely converted to a string
         :returns: None
         """
+        buf = _text_to_bytes_and_warn("buf", buf)
+        if _lib.SSL_CTX_set_session_id_context(
+                self._context,
+                buf,
+                len(buf),
+        ) != 1:
+            _raise_current_error()
```

Comment:
That said, in this case you could trigger the error by passing a session ID context value greater than `SSL_MAX_SSL_SESSION_ID_LENGTH`, which appears to be 32 (bytes?)